### PR TITLE
move fabric8-config to junit-5

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

**Comment by wind57** [comment]
@ryanjbaxter do not review this one for now, I need to provide extensive comments here.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/EventBasedConfigurationChangeDetectorTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/EventBasedConfigurationChangeDetectorTests.java:28]: the changes here are obvious: move to junit-5 + minor nitpick on a raw type.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigurationChangeDetectorTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigurationChangeDetectorTest.java:25]: trivial changes also : junit-5 + assertJ to be consistent

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java:47]: I hope this comment makes sense of why the configuration has moved in here from the common class. It is just the way `@EnableKubernetesMockClient` "injects" fields.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java:58]: mock client works slightly differently now and there is no need to mock paths, instead everything happens on the client itself. It is the same exact thing, logically.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/MultipleSecretsTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/MultipleSecretsTests.java:89]: refactor the anti-pattern away

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/reload/ConfigReloadAutoConfigurationTest.java:122]: `setup` method was slightly changed to get `Class<?>` as input, and I pass the `KubernetesClientTestConfiguration.class` configuration in there. everything else stays the same

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/MultipleConfigMapsTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/MultipleConfigMapsTests.java:89]: no need to mock paths anymore

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsFromFilePathsTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsFromFilePathsTests.java:68]: this is now injected by `@ExtendWith(SpringExtension` so no need for the server. the rest of the changes are trivial, imho, here.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsMixedTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsMixedTests.java:57]: this is now injected by `@EnableKubernetesMockClient(crud = true, https = false)` so no need for the server. the rest of the changes are trivial, imho, here.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsMixedTests.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsMixedTests.java:83]: no need to mock paths anymore

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsTest.java:57]: it tests the same thing, but where to find the "thing" to assert against is in a different place

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsTest.java:71]: another interesting change is that `KubernetesServer` used to return a `NamespacedKubernetesClient`, so you could do : `this.server.getClient().inNamespace(namespace)`, specifically call `inNamespace` with a certain `namespace`. `KubernetesClient` that is injected can't do that so I used it's default namespace `test`.

As such, I deleted `String namespace = "app-props";`. Fundamentally, the test does the same assertions.

Comment (wind57) [reply spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/ConfigMapsTest.java:71]: the same change is across all tests in this class

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java:53]: this was already fixed by me, waiting for a review. I've also asked if this will be back-ported. If it is (and you are OK with that, of course), I will upgrade the fabric8 libraries and revert this back.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/KubernetesConfigConfigurationTest.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/KubernetesConfigConfigurationTest.java:36]: why this is needed in this manner is explained in a similar class, I hope it makes sense.

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/KubernetesConfigTestBase.java
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/KubernetesConfigTestBase.java:39]: this method take one more argument now

**Comment by wind57** [comment]
@ryanjbaxter ready to review. The changes are not that big (but I am biased since I stayed quite a while in this code). The one change that I wish I could perform and really can't is to drop all `junit-4` dependencies (transitive ones), but because of [this issue](https://github.com/square/okhttp/issues/4667) that is internally used by `KubernetesMockServerExtension` that is declared on `@EnableKubernetesMockClient` - I can't. It's not a big deal, but I wish I could have done that...

**Comment by ryanjbaxter** [comment]
Can you mark it ready to review?

**Comment by wind57** [comment]
done

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java
Comment (ryanjbaxter) [reply spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java:53]: What version of fabric8 was it changed in?  Maybe we should make that change first, before making these?

**Review by ryanjbaxter — APPROVED** [review]

**Comment by ryanjbaxter** [comment]
Thanks for this, just one minor question!

File: spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java
Comment (wind57) [reply spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java:53]: this is still pending review from my PR [here](https://github.com/fabric8io/kubernetes-client/pull/3151). The good news is that it is targeted in `4.13.4` and not `5.x.x`. I am more than OK to wait for `4.13.4` release and only then merge this one.

That being said - I do have a minor question here. Why isn't this project moving towards fabric8 - `5.x.x`? thank you

Comment (ryanjbaxter) [reply spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java:53]: OK cool, its up to you.  If you don't mind going back and making that change I am happy to merge this now.

We will move to 5.x.x in our next release train.  We just don't want to make a major change like that and cause it to break people in a service release.

Comment (wind57) [reply spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java:53]: I will make the change in a separate PR, as soon as `4.13.4` is released. pinky-promise!

**Pull request merged**
